src/main/java/io/percy/selenium/Percy.java

```diff
@@ -96,65 +120,91 @@ public void snapshot(String name, List<Integer> widths) {
      * @param widths    The browser widths at which you want to take the snapshot.
      *                  In pixels.
      * @param minHeight The minimum height of the resulting snapshot. In pixels.
+     * @param enableJavaScript Enable JavaScript in the Percy rendering environment
      */
-    public void snapshot(String name, @Nullable List<Integer> widths, Integer minHeight) {
+    public void snapshot(String name, @Nullable List<Integer> widths, Integer minHeight, boolean enableJavaScript) {
+        String domSnapshot = "";
+
         if (percyAgentJs == null) {
             // This would happen if we couldn't load percy-agent.js in the constructor.
             LOGGER.log(Level.WARNING, "percy-agent.js is not available. Snapshotting is disabled.");
             return;
         }
+
         try {
             JavascriptExecutor jse = (JavascriptExecutor) driver;
             jse.executeScript(percyAgentJs);
-            jse.executeScript(buildSnapshotJS(name, widths, minHeight));
+            domSnapshot = (String) jse.executeScript(buildSnapshotJS());
         } catch (WebDriverException e) {
             // For some reason, the execution in the browser failed.
-            LOGGER.log(Level.WARNING, "Something went wrong attempting to take a snapshot: {}", e.getMessage());
```

Comment:
This logger was eating error messages and wouldn't print anything. Not sure why? Rolling with the 'ol reliable `System.out.println` for now